src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/onetimecode/OneTimeCodeService.kt

```diff
@@ -1,17 +1,42 @@
 package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.onetimecode
 
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.cjsm.CjsmService
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.config.ResourceNotFoundException
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.security.JwtService
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.security.SmokeTestConfig
 
 @Service
 class OneTimeCodeService(
   private val oneTimeCodeGenerator: OneTimeCodeGenerator,
   private val oneTimeCodeRepository: OneTimeCodeRepository,
   private val oneTimeCodeEmailSender: OneTimeCodeEmailSender,
+  private val jwtService: JwtService,
+  private val cjsmService: CjsmService,
+  private val smokeTestConfig: SmokeTestConfig,
 ) {
 
   fun createAndSendOneTimeCode(email: String, sessionId: String) {
     OneTimeCode(sessionId, oneTimeCodeGenerator.generateOneTimeCode(), email)
       .also { oneTimeCode -> oneTimeCodeRepository.save(oneTimeCode) }
       .also { oneTimeCode -> oneTimeCodeEmailSender.send(email, oneTimeCode.code) }
   }
+
+  fun verifyOneTimeCode(code: String, sessionId: String): String =
```

Comment:
In a previous PR (in fact its the method above), we added an endpoint and service method to create a one time code. Basically a OneTimeCode is an object containing the browser session id (effectively the PK), the one time code, and the users email address. This is stored in redis, and an email with the code sent to the user.

To verify the one time code the browser sends a POST request with the browser session id and the code. The verify service method (ie. this), retrieves the OneTimeCode from the redis repository by session ID, checks the code, then creates the JWT for the user and returns it.
If the redis cache does not contain the OneTimeCode keyed by session ID, an exception is thrown (returned to the client as a 404)

If the OneTimeCode is found but the code does not match, this is also (currently) returned as 404 via the same exception, but the OneTimeCode is not deleted from redis. This is to support the user to retry (ie. they mistyped the code) and to support a future story implementing a '3 strikes and out' type approach.